### PR TITLE
Tests & Travis & sentry/sdk ^3.0

### PR DESCRIPTION
**Pull request by martenb**

### CODE REVIEW

**Comment by martenb** [comment]
@rootpd Can you look at it, i have changed the initialization of sentry, which should be compatible with both ^2.1 and ^3.0.
Also i have added simple tests and travis config.
Tests are not working with --prefer-lowest dependecies because of some Nette BC, maybe you can fix in the future?

**Comment by martenb** [comment]
@rootpd ping

**Comment by rootpd** [comment]
I'm glad you had the chance and time to make the PR, and I'll get to that
eventually (review and test the changes).. But please remember, this is an
open source library, maintained in my spare time. There's no way I can just
drop the rest of my responsibilities and start working on this. And I
cannot just merge it without reviewing.




ut 13. 10. 2020 o 18:38 Martin Brettschneider <notifications@github.com>
napísal(a):

**Comment by martenb** [comment]
Ok, i have bad experiences with unmainatained libraries, so i was afraid, that this is same case, when you didnt respond.
I will use my fork now, until you have time.

**Comment by rootpd** [comment]
I did not forget about you, don't worry :). Just other things piled up.

File: composer.json [hunk 1]
```diff
@@ -13,18 +13,25 @@
   "require": {
     "ext-iconv":"*",
     "php": "^7.1",
-    "sentry/sdk": "^2.1",
+    "sentry/sdk": "^2.1 || ^3.0",
```
Comment (rootpd) [inline composer.json]: I know the library itself doesn't use anything internal from the Sentry SDK so it _shouldn't_ break, but is there any reason to keep `2.*` available? It feels like we're keeping a window for some kind of failure here. I'd probably go with `^3.0` directly.

What do you think?

**Review by rootpd — CHANGES_REQUESTED** [review]
Regarding the `--prefer-lowest` issues you mentioned. I've found out that the `Tracy\Logger` constructor changed since version `2.6`, and is causing the tests you've mentioned failing for any older version. It's almost 2-year old change, so I'm keen to just dump the old versions and require `tracy/tracy: ^2.6`. It still is a non-breaking change, so it should be fine.

If that sounds OK to you, please change it in this PR.

I've tested the newer version of Tracy and for PHP 7.4 I'm still seeing some `--prefer-lowest` issues in tests which would be fixed by updating `nette/di` to `3.0` and `nette/utils` to `3.0.1`, but that would be a breaking change so we'll keep it as it is.

Once these changes are done, I think we can squash and merge this.

File: composer.json (hunk 1 above)
Comment (martenb) [reply composer.json]: sentry/sdk 3.0 requires php 7.2, i think, we can support 7.1 for now...

with tracy.. i think, it should be fixed in tests more then drop support tracy 2.4, what do you think?

Comment (martenb) [reply composer.json]: or release version as it is and another with tracy ^2.6 and enabled prefer lowest tests

Comment (martenb) [reply composer.json]: i will try it again, to fix it in tests, but i have no luck with fixing it before..

Comment (martenb) [reply composer.json]: What do you think about this commit?

https://github.com/martenb/nette-sentry/commit/65952786760541e2cb39378cbedfafd19254fcf8

Comment (rootpd) [reply composer.json]: The default for logging directory in (https://github.com/martenb/nette-sentry/commit/65952786760541e2cb39378cbedfafd19254fcf8) seems OK to me, we could go with that.

Regarding the `sentry/sdk`, I'd still go with `3.0`, even if the bump to PHP 7.2 is required. PHP 7.1 has not been security-supported for over a year now (https://www.php.net/supported-versions.php) so it's actually a good idea to force people to upgrade. Even 7.2 will be done in a month so this is the least we could do here.

Comment (martenb) [reply composer.json]: ok, that makes sense. when we are changing libraries... why is there ext-iconv? it looks like, it is not used in library..?

Comment (rootpd) [reply composer.json]: Huh, I don't really remember. Let's remove it. I'll test the master in production before I tag the new version so we'll see whether it was necessary for something or not.

Comment (martenb) [reply composer.json]: uf, there was some depracations in nette, so it must look like this...

https://github.com/martenb/nette-sentry/commit/d7fce06489586f0f70272307d079fddcce2a1485

Comment (rootpd) [reply composer.json]: That's OK :+1:

Comment (martenb) [reply composer.json]: ok, i will rebase master

**Comment by rootpd** [comment]
Thanks a lot. One last thing to discuss. On PHP 7.4 and `--prefer-lowest` composer switch I'm getting this deprecation error:

```
> tester -C tests
 _____ ___  ___ _____ ___  ___
|_   _/ __)( __/_   _/ __)| _ )
  |_| \___ /___) |_| \___ |_|_\  v2.3.0

PHP 7.4.3 (cli) | php | 8 threads

F

-- FAILED: Extension.phpt
   Exited with error code 255 (expected 0)
   E_DEPRECATED: Function ReflectionType::__toString() is deprecated
   
   in src/Utils/Reflection.php(54) 
   in src/DI/Helpers.php(111) Nette\Utils\Reflection::getParameterType()
   in src/DI/ContainerBuilder.php(627) Nette\DI\Helpers::autowireArguments()
   in src/DI/ContainerBuilder.php(572) Nette\DI\ContainerBuilder->completeStatement()
   in src/DI/PhpGenerator.php(48) Nette\DI\ContainerBuilder->complete()
   in src/DI/Compiler.php(257) Nette\DI\PhpGenerator->generate()
   in src/DI/Compiler.php(177) Nette\DI\Compiler->generateCode()
   in src/DI/ContainerLoader.php(118) Nette\DI\Compiler->compile()
   in src/DI/ContainerLoader.php(82) Nette\DI\ContainerLoader->generate()
   in src/DI/ContainerLoader.php(47) Nette\DI\ContainerLoader->loadFile()
   in cases/DI/Extension.phpt(31) Nette\DI\ContainerLoader->load()
   in nunjuck/src/functions.php(18) {closure}()
   in cases/DI/Extension.phpt(50) test()
```

I think it's something that we probably don't need to handle since it's "just" the deprecation in PHP 7.4 and the only way how to solve it is to force `nette/utils: ^3.0.1` which would force Nette 3 everywhere.

The deprecation was fixed here: https://github.com/nette/utils/commit/fb3d29d1dbaa05660e68fa98bec1e43181202b91.

If you see a way how to circumvent it, let me know, otherwise I'll keep it as it is.

**Review by rootpd — APPROVED** [review]

**Comment by martenb** [comment]
I think, that whole nette 2.4 is not supossed to run on PHP 7.4

**Comment by rootpd** [comment]
Merged and tagged as `1.2.0`.

**Comment by martenb** [comment]
Perfect, thx

**Pull request merged**